mono/metadata/sgen-mono.c

```diff
@@ -1243,9 +1216,19 @@ create_allocator (int atype, gboolean slowpath)
 	 * another tls var holding its address instead.
 	 */
 
+	thread_info_var = mono_mb_add_local (mb, &mono_defaults.int_class->byval_arg);
+	mono_mb_emit_byte (mb, MONO_CUSTOM_PREFIX);
+	mono_mb_emit_byte (mb, CEE_MONO_TLS);
+	mono_mb_emit_i4 (mb, TLS_KEY_SGEN_THREAD_INFO);
```

Comment:
CEE_MONO_TLS is the 'fast implementation', its not available everywhere, i.e. in method-to-ir.c:

				ins = mono_create_tls_get (cfg, key);
				if (!ins) {
					if (cfg->compile_aot) {
						DISABLE_AOT (cfg);
						MONO_INST_NEW (cfg, ins, OP_TLS_GET);
						ins->dreg = alloc_preg (cfg);
						ins->type = STACK_PTR;
					} else {
						g_assert_not_reached ();
					}
				}